src/Network/Probe.php

```diff
@@ -127,11 +128,16 @@ private static function hostMeta($host)
 
 		if (!is_object($xrd)) {
 			$curlResult = Network::curl($url, false, ['timeout' => $xrd_timeout, 'accept_content' => 'application/xrd+xml']);
+			$connection_error = ($curlResult->getErrorNumber() == CURLE_COULDNT_CONNECT) || ($curlResult->getReturnCode() == 0);
 			if ($curlResult->isTimeout()) {
 				Logger::info('Probing timeout', ['url' => $url], Logger::DEBUG);
 				self::$istimeout = true;
 				return false;
+			} elseif ($connection_error && $ssl_connection_error) {
```

Comment:
Why don't you check `$curlResult->isSuccess()` here as well like for the SSL curl call?

Reply:
This is superfluous here, since "isSuccess" is only checking for the return code which is checked for the "connection_error" variable anyway.

Reply:
The `$connection_error && $ssl_connection_error` seems very specific. I understand it is if both requests in connection errors.

But what if the second request fails with a return code > 0 (say 404)?

Reply:
404 would be totally okay. This would mean that there is some server responding. Then we can continue the probing. But when there is a connection or timeout issue on both SSL and non SSL then we don't need to perform any more probes.

Reply:
But we would retrieve the body and try to import XRD data even on 404 according to this code.

Reply:
We would try the default webfinger path. And we would try to fetch the profile via the ActivityPub-methods - which would be totally okay. A working ActivityPub implementation doesn't need any "hostmeta" nor a "webfinger". So we mustn't quit here on some "404".

Reply:
Gotcha.